File: leetcode-python-go/1796. 字符串中第二大的数字/main.py
s = "dfa12321afd"

def secondHighest(s: str) -> int:
    first = second = -1
    for c in s:
        if c.isdigit():
            num = int(c)
            if num > first:
                second = first
                first = num
            elif second < num < first:
                second = num
    return second

print(secondHighest(s))

File: leetcode-python-go/1796. 字符串中第二大的数字/main.go
package main

import (
	"fmt"
	"unicode"
)

func secondHighest(s string) int {
	first, second := -1, -1
	for _, c := range s {
		if unicode.IsDigit(c) {
			num := int(c - '0')
			if num > first {
				second = first
				first = num
			} else if second < num && num < first {
				second = num
			}
		}
	}
	return second
}

func main() {
	s := "dfa12321afd"
	results := secondHighest(s)
	fmt.Println(results)
}
